packages/eservice-template-process/src/services/eserviceTemplateService.ts

```diff
@@ -292,6 +295,99 @@ export function eserviceTemplateServiceBuilder(
       await repository.createEvent(event);
     },
 
+    async publishEServiceTemplateVersion(
+      eserviceTemplateId: EServiceTemplateId,
+      eserviceTemplateVersionId: EServiceTemplateVersionId,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<void> {
+      logger.info(
+        `Publishing e-service template version ${eserviceTemplateVersionId} for EService ${eserviceTemplateId}`
+      );
+
+      const eserviceTemplate = await retrieveEServiceTemplate(
+        eserviceTemplateId,
+        readModelService
+      );
+
+      assertRequesterEServiceTemplateCreator(
+        eserviceTemplate.data.creatorId,
+        authData
+      );
+
+      const eserviceTemplateVersion = retrieveEServiceTemplateVersion(
+        eserviceTemplateVersionId,
+        eserviceTemplate.data
+      );
+
+      if (
+        eserviceTemplateVersion.state !== eserviceTemplateVersionState.draft
+      ) {
+        throw notValidEServiceTemplateVersionState(
+          eserviceTemplateVersionId,
+          eserviceTemplateVersion.state
+        );
+      }
+
+      if (eserviceTemplateVersion.interface === undefined) {
+        throw missingTemplateVersionInterface(
+          eserviceTemplateId,
+          eserviceTemplateVersionId
+        );
+      }
+
+      const tenant = await retrieveTenant(
+        eserviceTemplate.data.creatorId,
+        readModelService
+      );
+      assertTenantKindExists(tenant);
+
+      if (
+        eserviceTemplate.data.mode === eserviceMode.receive &&
+        (eserviceTemplate.data.riskAnalysis.length === 0 ||
+          eserviceTemplate.data.riskAnalysis.some((ra) => {
+            const result = validateRiskAnalysis(
+              riskAnalysisFormToRiskAnalysisFormToValidate(ra.riskAnalysisForm),
+              true,
+              tenant.kind
+            );
+
+            return result.type === "invalid";
+          }))
+      ) {
+        // should we return the validation issues or it's ok to just return an empty issues array?
+        throw riskAnalysisValidationFailed([]);
```

Comment:
Should we return the validation issues or is it's ok just to return an empty issues array?

Reply:
mmh... Why not?

Reply:
I’d suggest a fail-fast approach, I'm okkay with the first point